pallets/rad-claims/src/lib.rs

```diff
@@ -0,0 +1,577 @@
+    // Pallet unsigned transactions validation
+    // ------------------------------------------------------------------------
+
+    #[pallet::validate_unsigned]
+	impl<T: Config> ValidateUnsigned for Pallet<T> {
+            
+        type Call = Call<T>;
+
+        /// Validate unsigned transactions
+        ///
+        /// Unsigned transactions are generally disallowed. However, since a contributor
+        /// claiming a reward payout may not have the necessary tokens on the parachain to
+        /// pay the fees of the claim, the [`claim_reward`] transactions must be 
+        /// unsigned.
+        /// Here, we make sure such unsigned, and remember, feeless unsigned transactions
+        /// can be used for malicious spams or Deny of Service (DoS) attacks.
+        fn validate_unsigned(
+            _source: TransactionSource, 
+            call: &Self::Call
+        ) -> TransactionValidity {
+            if let Call::claim(account_id, amount, sorted_hashes) = call {
+                // Check that proofs are valid with a root that exists in the root hash storage
+                if Self::verify_proofs(account_id, amount, sorted_hashes.into()) {
+                    return ValidTransaction::with_tag_prefix("RadClaims")
+                        .priority(T::UnsignedPriority::get())
+                        .and_provides((account_id, amount, sorted_hashes))
+                        .longevity(TryInto::<u64>::try_into(
+                            T::Longevity::get())
+                            .unwrap_or(64_u64))
+                        .propagate(true)
+                        .build()
+                } else {
+                    return InvalidTransaction::BadProof.into();
+                }
+            }
+    
+            InvalidTransaction::Call.into()
+        }
+    }
+} // end of 'pallet' module
+
+
+// ----------------------------------------------------------------------------
+// Pallet implementation block
+// ----------------------------------------------------------------------------
+
+// Pallet implementation block.
+//
+// This main implementation block contains two categories of functions, namely:
+// - Public functions: These are functions that are `pub` and generally fall into 
+//   inspector functions that do not write to storage and operation functions that do.
+// - Private functions: These are private helpers or utilities that cannot be called 
+//   from other pallets.
+impl<T: Config> Pallet<T> {
+   
+    /// Return the account identifier of the RAD claims pallet.
+	///
+	/// This actually does computation. If you need to keep using it, then make
+	/// sure you cache the value and only call this once.
+	pub fn account_id() -> T::AccountId {
+	  T::PalletId::get().into_account()
+	}
+
+    /// TODO: need function documentation
+    /// Hash a:b if a < b, else b:a. Uses the runtime module's hasher.
+    pub fn sorted_hash_of(a: &T::Hash, b: &T::Hash) -> T::Hash {
```

Comment:
I think we have a similar function on proofs as well. Maybe we can use that after the proofs are moved over here